CryptoPrices/Sources/Data/Sources/DataTestHelpers/Mock/MockCoinAPI.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  MockCoinAPI.swift
+//  Data
+//
+//  Created by Doan Thieu on 09/12/2022.
+//
+
+import NetworkCore
+
+// TODO: - Set up code generator for generating mock types (e.g. Sourcery)
+
+public class MockCoinAPI: CoinAPIProtocol {
+
+    public var myCoinsReturnValue: Result<[APICoin], Error>?
+
+    public init(myCoinsReturnValue: Result<[APICoin], Error>? = nil) {
+        self.myCoinsReturnValue = myCoinsReturnValue
+    }
+
+    public func myCoins() async throws -> [APICoin] {
+        guard let myCoinsReturnValue = myCoinsReturnValue else {
```

Comment:
FYI, we could shorten it like this in swift 5.7 :D:
```suggestion
        guard let myCoinsReturnValuee else {
```

Reply:
Actually, I intend to make this app compatible with Swift 5.5 (Xcode 13.2.1 if I remember correctly). You can see in all the local package's manifest (`Package.swift` file), the very first lines are:
```
// swift-tools-version: 5.5
// The swift-tools-version declares the minimum version of Swift required to build this package.
```

So while this is nice and improved syntax, it won't compile with Xcode with versions of `13.2.1` ~ < `14.0.0`.
Later if we decide to lift up the whole project to the latest Swift version, we can have a dedicated to do so. It's because it's not simple as just using the newer Xcode. There are chances that third-party libraries won't be compatible 🙏